clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -639,19 +645,22 @@ private void maybeUpdate(long now, Node node) {
                 ClientRequest clientRequest = request(now, nodeConnectionId, metadataRequest);
                 log.debug("Sending metadata request {} to node {}", metadataRequest, node.id());
                 doSend(clientRequest, now);
-            } else if (connectionStates.canConnect(nodeConnectionId, now)) {
+            } else if ((lastConnectAttemptNodeId != null && isConnectionClosed(lastConnectAttemptNodeId) ||
```

Comment:
hmm, is it possible?
guess it doesn't work in following case?

1. Have two nodes.  Node-A(disconnected), Node-B(connected, in-flight-requests is 1)
2. `leastLoadedNode()` returns Node-A which isn't connected yet. Initiate connection to Node-A. 
3. On the next `poll()`, a response from Node-B received so the in-flight-requests is now 0.
4. On next `maybeUpdate()` `leastLoadedNode()` returns Node-B.